### ISSUE TITLE
bug(iOS): In-app suggestion banner clipped in iOS 15

### ISSUE THREAD

**Issue by bharanidharanj** [keymanapp/keyman#6595]
**Describe the bug**
Suggesting Words appears partially in iPhone 13 Pro Simulator after installing Keyman 15.0.243 Beta build

**To Reproduce**
1. Download the latest Keyman Beta build 15.0.243 for iOS.
2. Install Keyman in iPhone 13 Pro Simulator.
3. Click on the Keyman icon from the Display.
4. Here, I noticed that the Suggesting Words appears in the banner appearing partially. 

**Expected behavior**
Suggesting Words should appear (without any glitch) in the banner correctly.

**Screenshots**
<img width="300" src="https://user-images.githubusercontent.com/19683143/167416406-6e442560-46ea-467d-a0b8-b000d527d38b.png"> 

---

**Keyman for iPhone/iPad/Android:**
 - Device: iPhone 13 Pro Simulator
 - OS:  iOS 15.4
 - Keyman version 15.0.243-beta build
 - Target application: Keyman 

---

**Keyboard**
 - Keyboard name: EuroLatin (SIL)
 - Keyboard version: 1.9.6
 - Language name: EuroLatin

**Additional context**
Note: 
1. I noticed that this happens only for the first time after opening the Keyman. If I close and reopen the Keyman, I am not seeing this problem. 
2. It would be better to retest the same in real iPhone device.

**Comment by bharanidharanj** [comment]
Tested this issue in landscape mode in iPhone 13 Pro Simulator and I was able to reproduce the same. 

<img width="300" src="https://user-images.githubusercontent.com/19683143/167862542-520794a2-b9ce-46ad-85cc-65946dbb1e8b.png">

**Comment by MakaraSok** [comment]
@bharanidharanj I've tried the build artifact from [here](https://build.palaso.org/buildConfiguration/Keyman_iOS_TestPullRequests/317677?buildTab=artifacts#%2Fupload) with the same version you indicated (v15.0.243-beta), but I don't see the issue in the simulator:


<img width="300" alt="image" src="https://user-images.githubusercontent.com/28331388/167991878-805cdad7-7b31-4f73-a8cf-b181c9430601.png">


On physical devices via TestFlight, v15.0.243 published on 2022-04-29 for pr #6562 was used to test this, but everything looks OK on the banner.

| iPod touch 7th gen | iPad mini 6 |
| --- | --- |
| <img width="200" alt="image" src="https://user-images.githubusercontent.com/28331388/167992114-0caedd95-9c4c-4f11-a196-07105998e263.png"> | <img width="200" alt="image" src="https://user-images.githubusercontent.com/28331388/167992131-62f6c4d8-11a0-4ba3-be4b-448661274e76.png"> |

**Comment by sgschantz** [comment]
**This happens in iOS 15 only.**  Was able to reproduce with iOS 15.4 on simulator for 

- iPhone 13 Pro
- 12 mini
- 11 (text almost completely gone)
- iPad Pro 5th gen

**Was unable to reproduce with iOS 14.4** on simulator for iPhone 12 mini,  iPhone 11

Steps to reproduce:
1. install Keyman on simulator
2. start Keyman: you may see clipped keyboard flash on the screen before it is obscured by Get Started dialog
3. click the 'Don't show again' switch on the Get Started dialog
4. click Done to dismiss Get Started dialog
5. kill the app
6. start the app again
7. when the app opens, the top half of the suggestion banner is clipped so that the words are barely visible

If you go to the home screen and then open the app again, the view will be correct.
However, if you kill the app and open it again, the suggestion banner will be clipped again.

**Comment by jahorton** [comment]
After some experimentation, I've noticed something interesting.

I can reproduce very similar behavior on _unaffected devices_ by disabling the "safe area" constraint at the bottom of the keyboard that is currently applied in `InputViewController`'s custom `InputView`.  We've been applying this constraint so that the `KeymanWebView` doesn't need to worry about safe area nonsense or anything else - we can tell it a size and have that stick.

When that `KeymanWebView`-external constraint is disabled... the `WKWebView` itself will _internally_ apply a very similar constraint, pushing the keyboard out of the, uh, "'unsafe' area" - and particularly, even on devices that don't experience the reported bug.

What I noticed when working with @sgschantz - the behavior we're seeing appears to indicate that _both_ an "external" constraint - the one we intend - _and_ an "internal" constraint _that we don't intend_ are both being applied.  It's as if for affected devices, the WebView is autoapplying a safe area constraint to the internal page despite not being within the "'unsafe' area" after layout.  So we're getting what appears to be a double-application of the same constraint.

We also noticed that this bug does not affect iOS 15 devices without a notch or similar "'unsafe' area" at the bottom, such as the newer iPhone SE phones.  Naturally, because there is no "'unsafe' area", double-application of an adjustment of zero is... an adjustment of zero.  So everything appears as it should there.

One other (minor) note:  an interesting detail on the JS / Web side when this bug occurs...

`window.innerHeight` will generally equal `oskHeight - <unsafe area height>`, where `oskHeight` is the height given to `KeymanWebView`.  The part after the 'minus' should always be effectively zero, as far as KMW is concerned, due to effects from the constraint set within `InputViewController`'s view.

------

My initial thought, given these observations, was that perhaps layout is happening before our ("external") constraint is applied, thus the Safari-based WebView notes a need to apply its own constraint.  Then, hypothetically, our constraint then goes active later and bumps the WebView up, resulting in the appearance of a double constraint.

_Except_... when we add calls to `fixLayout()` / `layoutIfNeeded()` at later points in the display cycle... the bug still remains.  Maybe it's that we still need to find a "good" place to make a post-adjustment?  Either way, we never needed to do this before.
